interviews/arrow-func-constructor/answers/alexafcode/index.md

```diff
@@ -0,0 +1,23 @@
+﻿В чём плюс использования стрелочных функций для методов в конструкторе?
+
+Основным плюсом использования стрелочных функций для методов в конструкторе является то, что `this` будет установлено во время создания функции и `this` не будет изменяться. Cледовательно, когда конструктор используется для создания нового объекта, `this` всегда будет ссылаться на этот объект.
```

Comment:
Давай попробуем немного переформулировать. Текущий вариант вызывает несколько вопросов, например:
Когда и где будет установлен этот this? 
Если this устанавливается в момент создания функции, то какой именно? та что конструктор или та которая стрелочная? 

Я бы попробовала описать как оно устроено на самом деле. 
Примерно так: любая функция выполняется и создается в рамках некоторого _контекста выполнения_
Внутри этого контекста может быть и может и не быть this-биндинга.
Если this-биндинг есть (в конструкторе он есть, this внутри функции связывается с this из текущего (лексического) контекста выполнения)

Вот точная цитата из специ
```
An [ArrowFunction](https://tc39.es/ecma262/#prod-ArrowFunction) does not define local bindings for arguments, super, this, or new.target. Any reference to arguments, super, this, or new.target within an [ArrowFunction](https://tc39.es/ecma262/#prod-ArrowFunction) must resolve to a binding in a lexically enclosing environment. Typically this will be the Function Environment of an immediately enclosing function. Even though an [ArrowFunction](https://tc39.es/ecma262/#prod-ArrowFunction) may contain references to super, the [function object](https://tc39.es/ecma262/#function-object) created in step [5](https://tc39.es/ecma262/#step-arrowfunction-evaluation-functioncreate) is not made into a method by performing [MakeMethod](https://tc39.es/ecma262/#sec-makemethod). An [ArrowFunction](https://tc39.es/ecma262/#prod-ArrowFunction) that references super is always contained within a non-[ArrowFunction](https://tc39.es/ecma262/#prod-ArrowFunction) and the necessary state to implement super is accessible via the env that is captured by the [function object](https://tc39.es/ecma262/#function-object) of the [ArrowFunction](https://tc39.es/ecma262/#prod-ArrowFunction).```